esp/GBPlay/main/tasks/socket_manager.c

```diff
@@ -0,0 +1,148 @@
+#include <freertos/FreeRTOS.h>
+#include <freertos/event_groups.h>
+#include <freertos/task.h>
+
+#include <esp_event.h>
+#include <esp_log.h>
+
+#include "../hardware/spi.h"
+#include "../hardware/storage.h"
+#include "../hardware/wifi.h"
+#include "socket.h"
+
+#define TASK_NAME "socket-manager"
+
+#define CONNECTION_TIMEOUT_MS 10000
+
+#define SERVER_HOST_STORAGE_KEY "server_host"
+#define SERVER_PORT_STORAGE_KEY "server_port"
+
+#define DEFAULT_SERVER_HOST "192.168.0.115"
```

Comment:
Use `set-value server_host <my_host>` and `set-value server_port <my_port>` to change the default server configuration (see `commands.c`). This only needs to be done once. You can undo it with `delete-value <value_key>`.